### PR TITLE
Ability to create users by superadmin

### PR DESCRIPTION
**Pull request by shlok007**

### CODE REVIEW

**Comment by shlok007** [comment]
mail

This is the preview of the message the user would receive.

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -22,8 +22,11 @@ def create
       create_params = resource_params
       create_params.delete(:is_superadmin)
       resource = resource_class.new(create_params)
-
+      resource.password=resource.generate_password
```
Comment (shubhpatel108) [inline app/controllers/admin/users_controller.rb]: space before and after = in assignment statement

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -22,8 +22,11 @@ def create
...
+      resource.password_confirmation=resource.password
+      resource.skip_confirmation!
       if resource.save
```
Comment (shubhpatel108) [inline app/controllers/admin/users_controller.rb:29]: space before the if

File: app/views/user_mailer/send_password.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  </head>
+  <body>
+    <h1>Welcome to Book-a-Nook!</h1>
+    <p>
+      You have been successfully signed up to bookanook.com<br><br>
+      Your username : <b> <%= @user.email %></b><br>
+      Your password : <b> <%= @user.password %></b><br>
+    </p>
+    <p>
+      To login to the site, just follow this link: http://bookanook.com
```
Comment (shubhpatel108) [inline app/views/user_mailer/send_password.html.erb]: Don't hard code the link. Use routes helper, for e.g. <%= password_change_path %>

File: app/views/user_mailer/send_password.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html>
```
Comment (shubhpatel108) [inline app/views/user_mailer/send_password.html.erb]: Not sure you will need the main tags as you are rendering them from layouts/mailer.html.erb.

File: app/models/user.rb [hunk 5]
```diff
@@ -19,6 +19,10 @@ def superadmin?
     return has_role? :superadmin
   end
 
+  def generate_password
+    ((1..9).to_a+('a'..'z').to_a).shuffle[0,10].join
```
Comment (shubhpatel108) [inline app/models/user.rb]: Are you sure this is the best way to generate random password? Do confirm it once.
Also, a space after comma and before & after any operator (here, +).

File: spec/mailers/user_mailer_spec.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+require "rails_helper"
+
+RSpec.describe UserMailer, type: :mailer do
+  before(:each) do
+    FactoryGirl.create :user
+  end
+
+  it "should deliver mail containing password" do
+    expect{UserMailer.send_password(User.last).deliver_now}.to change {ActionMailer::Base.deliveries.count}.by(1)
```
Comment (shubhpatel108) [inline spec/mailers/user_mailer_spec.rb]: Use curve brackets instead of the curly ones. 
Really great thought to test the mailers too!

Comment (shlok007) [inline spec/mailers/user_mailer_spec.rb]: Thanx. Can you please have a look at this? [stackoverflow.com/questions/25227196/expected-count-to-have-changed-by-1-but-was-not-given-a-block](http://stackoverflow.com/questions/25227196/expected-count-to-have-changed-by-1-but-was-not-given-a-block) . Even when I use curve brackets in the expect part it generates an error.

File: app/models/user.rb (hunk 5 above)
Comment (shlok007) [inline app/models/user.rb]: It's just generating a random string. I'll add the capital letters too to make it more secure.

File: app/views/user_mailer/send_password.html.erb (hunk 4 above)
Comment (shlok007) [inline app/views/user_mailer/send_password.html.erb]: I'll remove the main tags :+1:

File: spec/mailers/user_mailer_spec.rb (hunk 6 above)
Comment (shubhpatel108) [inline spec/mailers/user_mailer_spec.rb]: Oh sorry, didn't know that. Use curly ones then. Add spaces before after them.

Comment (shlok007) [inline spec/mailers/user_mailer_spec.rb]: Sure. :+1:

**Comment by shlok007** [comment]
Here is the updated message:
mail message

File: app/mailers/user_mailer.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class UserMailer < ApplicationMailer
+	
+  def send_password(new_user)
+    @user=new_user
```
Comment (shubhpatel108) [inline app/mailers/user_mailer.rb]: spaces before and after =

File: app/mailers/user_mailer.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+    mail(to: @user.email,subject: 'Welcome to bookanook!')
```
Comment (shubhpatel108) [inline app/mailers/user_mailer.rb]: spaces after ,

File: app/controllers/admin/users_controller.rb [hunk 9]
```diff
@@ -22,8 +22,12 @@ def create
       create_params = resource_params
       create_params.delete(:is_superadmin)
       resource = resource_class.new(create_params)
-
+      resource.password = resource.generate_password
+      resource.password_confirmation = resource.password
+      resource.skip_confirmation!
+
```
Comment (shubhpatel108) [inline app/controllers/admin/users_controller.rb]: An empty line should not contain spaces. An example of the rule - no line should have trailing spaces.

Comment (shlok007) [inline app/controllers/admin/users_controller.rb]: Thank you. I'll keep that in mind.

**Pull request merged**